source/extensions/filters/udp/dns_filter/dns_parser.h

```diff
@@ -352,9 +345,12 @@ class DnsMessageParser : public Logger::Loggable<Logger::Id::filter> {
                             Network::Address::InstanceConstSharedPtr ipaddr);
 
   /**
+   * @param context the query context for which we are querying the response code
    * @return uint16_t the response code flag value from a parsed dns object
    */
-  uint16_t getQueryResponseCode() { return static_cast<uint16_t>(header_.flags.rcode); }
```

Comment:
seems that if the caller has the context, this method should just be moved to the DnsQueryContext class

Reply:
Good call.  Let me update this.